tests/tests_unit/test_utils/test_concurrency.py

```diff
@@ -0,0 +1,31 @@
+from concurrent.futures import ThreadPoolExecutor
+
+from cognite.client.utils._concurrency import (
+    ConcurrencySettings,
+    MainThreadExecutor,
+    execute_tasks_concurrently,
+    get_executor,
+)
+
+
+class TestExecutor:
+    def test_set_and_get_executor(self) -> None:
+        executor = get_executor(1)
+        assert isinstance(executor, ThreadPoolExecutor)
+
+        ConcurrencySettings.executor_type = "mainthread"
```

Comment:
Will this have a side effect for later tests? (`monkeypatch.setattr`). Btw, sorry for not using a review, thought I'd only have one comment 😂

Reply:
no worries 😄 and yes. that's a good catch.